src/core/jit/synth/synth.rs

```diff
@@ -1,29 +1,49 @@
-use async_graphql::Name;
-use async_graphql_value::ConstValue;
-use indexmap::IndexMap;
-
 use super::super::Result;
-use super::Synthesizer;
-use crate::core::jit::model::{Field, Nested};
-use crate::core::jit::store::{Data, Store};
-use crate::core::jit::{DataPath, ExecutionPlan, Variable, Variables};
-use crate::core::json::JsonLike;
-
-pub struct Synth {
-    selection: Vec<Field<Nested<ConstValue>, ConstValue>>,
-    store: Store<Result<ConstValue>>,
-    variables: Variables<ConstValue>,
+use crate::core::jit::model::{ExecutionPlan, Field, Nested, Variable, Variables};
+use crate::core::jit::store::{Data, DataPath, Store};
+use crate::core::jit::synth::Synthesizer;
+use crate::core::json::{JsonLikeOwned, JsonObjectLike};
+
+// TODO: rename
+pub struct AlsoSynth<Value: JsonLikeOwned> {
```

Comment:
```suggestion
pub struct AlsoSynth<Value> {
```

Add bounds to impl always and not to constructors.